File: Ground_Booking_App/src/main/java/com/example/demo/entities/Ground.java
package com.example.demo.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "ground")
public class Ground {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "gId") Integer gId;

    @Column(name = "gName")
    private String gName;

    @Column(name = "gDescription")
    private String gDescription;

    @Column(name = "address")
    private String address;

    @ManyToOne
    @JoinColumn(name = "cId", referencedColumnName = "cId")
    private City city;

    @ManyToOne
    @JoinColumn(name = "uId", referencedColumnName = "uId")
    private User user;

    @ManyToOne
    @JoinColumn(name = "sId", referencedColumnName = "sId")
    private Sport sport;

    @Column(name = "gStatus")
    private String gStatus;

    @Column(name = "gImages")
    private String gImages;

    public Ground() {
    }

    // Ground variable getters and setters
    public Integer getgId() {
        return gId;
    }

    public void setgId(Integer gId) {
        this.gId = gId;
    }

    public String getgName() {
        return gName;
    }

    public void setgName(String gName) {
        this.gName = gName;
    }

    public String getgDescription() {
        return gDescription;
    }

    public void setgDescription(String gDescription) {
        this.gDescription = gDescription;
    }
    
    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getgStatus() {
        return gStatus;
    }

    public void setgStatus(String gStatus) {
        this.gStatus = gStatus;
    }

    public String getgImages() {
        return gImages;
    }

    public void setgImages(String gImages) {
        this.gImages = gImages;
    }
    
    // Related entity getters and setters
    public City getCity() {
        return city;
    }

    public void setCity(City city) {
        this.city = city;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Sport getSport() {
        return sport;
    }

    public void setSport(Sport sport) {
        this.sport = sport;
    }
}

File: GroundBooking.AdminAPI/GroundBooking.AdminAPI/Models/Ground.cs
using System;
using System.Collections.Generic;

namespace GroundBooking.AdminAPI.Models;

public partial class Ground
{
    public int GId { get; set; }

    public string GName { get; set; } = null!;

    public string GDescription { get; set; } = null!;

    public string Address { get; set; } = null!;

    public int CId { get; set; }

    public int UId { get; set; }

    public int SId { get; set; }

    public string? GStatus { get; set; }

    public string GImages { get; set; } = null!;

    public virtual ICollection<Booking> Booking { get; set; } = new List<Booking>();

    public virtual City CIdNavigation { get; set; } = null!;

    public virtual ICollection<Request> Request { get; set; } = new List<Request>();

    public virtual Sport SIdNavigation { get; set; } = null!;

    public virtual Users UIdNavigation { get; set; } = null!;
}
